### PR TITLE
Throttle block acceptance on specified conditions

### LINKED ISSUE #1331: Slow down consensus agreement time suitably
**Describe what you want implemented** 
AcceptBlock procedure should execute for not less than 1s (configurable sleep). The synthetic delay should not be applied in out-sync mode.

**Describe "Why" this is needed**
This should slow down consensus until more Provisioners join the network or Rusk grpc calls need more than 1s to complete.

**Describe alternatives you've considered**
FetchOrTimeout in Block generator does the trick too. However it works only if mempool is empty and it does not depend on grpc calls execution time. 

**Additional context**
The additional delay should guarantee a healthy consensus time not less than 5s.

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #1331

### CODE REVIEW

File: pkg/core/chain/chain.go [hunk 1]
```diff
@@ -349,6 +372,23 @@ func (c *Chain) acceptSuccessiveBlock(blk block.Block, kadcastHeight byte) error
...
+	// node is in out-of-sync mode.
+	// block time is higher that ConsensusTimeThreshold
+	if prevBlockTimestamp+config.ConsensusTimeThreshold > blk.Header.Timestamp {
+		maxDelayMilli := config.Get().Consensus.ThrottleMilli
+		if maxDelayMilli == 0 {
+			maxDelayMilli = 2000
+		}
+
+		if d, err := throttle(startTime, maxDelayMilli); err == nil {
+			log.WithField("height", blk.Header.Height).WithField("sleep_for", d.String()).Trace("throttled")
```
Comment (herr-seppia) [inline pkg/core/chain/chain.go:388]: Should `d` be used to wait? Cannot figure out where this duration is used

File: pkg/core/chain/chain.go [hunk 2]
```diff
@@ -336,11 +337,33 @@ func (c *Chain) ProcessSyncTimerExpired(strPeerAddr string) error {
...
+}
+
 // acceptSuccessiveBlock will accept a block which directly follows the chain
 // tip, and advertises it to the node's peers.
 func (c *Chain) acceptSuccessiveBlock(blk block.Block, kadcastHeight byte) error {
+	startTime := time.Now().UnixMilli()
+
 	log.WithField("height", blk.Header.Height).Trace("accepting succeeding block")
 
+	prevBlockTimestamp := c.tip.Header.Timestamp
```
Comment (herr-seppia) [inline pkg/core/chain/chain.go:365]: I'm a bit concerned to use timestamps in environment where no ntp is configured

Could the received instant be used for this scenario?

File: pkg/core/chain/chain.go (hunk 1 above)
Comment (goshawk-3) [reply pkg/core/chain/chain.go:389]: Good catch. Forgot to commit it. It's in throttle function.

File: pkg/core/chain/chain.go (hunk 2 above)
Comment (goshawk-3) [reply pkg/core/chain/chain.go:366]: Timestamp in block header is a field being validated. It's guaranteed that it's bigger than prevBlock timestamp.

Comment (herr-seppia) [reply pkg/core/chain/chain.go:366]: Ok, in this case I open this followUp issue to tackle some odd corner cases

https://github.com/dusk-network/dusk-blockchain/issues/1334

**Review by herr-seppia — APPROVED** [review]
LGTM

**Pull request merged**